pyproject.toml

```diff
@@ -38,7 +38,8 @@ dependencies = [
   # direct dependencies
   "build >= 1.0.0",
   "click >= 8",
-  "pip >= 22.2",
+  "pip >= 23.1; python_version >= '3.12'",
+  "pip >= 22.2; python_version < '3.12'",
```

Comment:
Is the conditional required here? Without it, would the rule supersede the previous, or both apply?

Reply:
I think that not all pip versions would merge these properly otherwise. But I haven't checked in a while. FWIW, this is the typical idiom people tend to follow.